packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -436,9 +436,15 @@ rocksdb::Status ArbCore::reorgToMessageOrBefore(
         pending_checkpoint = Checkpoint{};
     }
 
+    auto status =
```

Comment:
Seems like there might be an off by one issue here. This checkpoint is still valid so it should be `pending_checkpoint.block_height` I think